mailscanner/functions.php

```diff
@@ -2972,6 +2972,46 @@ function ldap_authenticate($username, $password)
 }
 
 /**
+ * @param string $username
+ * @param string $password
+ * @return null|string
+ */
+function imap_authenticate($username, $password)
+{
+    $username = strtolower($username);
+    $email = $username;
+
+    if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+        //user has no mail but it is required for mailwatch
+        return null;
+    }
+
+    if ($username !== '' && $password !== '') {
+        $mbox = imap_open(IMAP_HOST, $username, $password, NULL, 0);
+
+        if ( false === $mbox ) {
+            //auth faild
+            return null;
+        }
+
+        $sql = sprintf('SELECT username FROM users WHERE username = %s', quote_smart($email));
+        $sth = dbquery($sql);
+        if ($sth->num_rows === 0) {
```

Comment:
Add a `IMAP_AUTOCREATE_VALID_USER` boolean constant to enable this behaviour, set it false by default